slides/hash_functions.tex

```diff
@@ -0,0 +1,242 @@
+\begin{center}
+CH\color{focus}56\color{black}04835012345678009 vs. CH\color{focus}56\color{black}048350123456\color{focus}877\color{black}09
+\end{center}}
+
+
+$\Rightarrow$ For security purposes, simple hashes are of limited use
+	
+\end{frame}
+%%%	
+
+%%%
+\begin{frame}{Cryptographic Hash Functions}
+
+Additional criteria:
+	\begin{enumerate}
+		\item Approximately uniform hash value distribution
+		\item Quick to compute for any given pre-image
+		\item Trap-door: Infeasible to generate pre-image from hash value
+		\item Avalanche effect: Small change in input results in totally different output
+		\item Very low collision probability: Unlikely that two pre-images generate the same hash value.
+	\end{enumerate}
+	\vspace{1em}
+
+\uncover<1->{In Bitcoin context, the functions \color{focus}SHA2.256 \color{black} (in short SHA256) and \color{focus}RIPEMD160 \color{black}are used. Both satisfy the above criteria.}
+	
+\end{frame}
+%%%	
+
+%%%
+\begin{frame}{Avalanche Effect with SHA2}
+
+\begin{center}
+$SHA256(<$\textit{pre-image}$>)$
+\end{center}
+
+This is the pre-image
+
+$\Rightarrow$ \footnotesize 5e13fffedd642aaddea7872463fd44d8b8a336cb822ebd4c12d9e6282c88cea8 \normalsize
+\vspace{1em}
+
+\color{focus}t\color{black}his is the pre-image
+
+$\Rightarrow$ \footnotesize \color{focus}aba56d7c281f8d\color{black}a\color{focus}cb1076baed182538d2ce21494075f7377982bf9d5d0\color{black}8\color{focus}e6ccf\color{black} \normalsize
+\vspace{1em}
+
+\begin{columns}[T]
+	\begin{column}{0.35\textwidth}
+		\includegraphics[width = 4 cm, frame]{../assets/images/manual_hashing_video.png}
+	\end{column} %\hfill
+	\begin{column}{0.65\textwidth}
+		\begin{itemize}
+			\item Nonlinarity due to choice, majority, mod, rotation and shifting. operations
+			\item Efficient for computers vs. 0.67 hashes / day by hand.
+			\item \link \href{https://www.youtube.com/watch?v=y3dqhixzGVo}{Manual calculation video}.
+		\end{itemize}
+	\end{column}
+\end{columns}
+
+	
+\end{frame}
+%%%	
+
+%%%
+\begin{frame}{Low Collision Probability}
+
+Due to the fixed size of $h$, the corresponding hash function $H()$ can only produce a finite set of distinct hashes.
+\vspace{1em}
+
+SHA256:
+	\begin{itemize}
+		\item Almost all known combinations of 256 Bits, i.e.,  $2^{256}$ $h$
+		\item In base 10, this is any number between 1 and 115'792'089' 237'316'195'423'570'985'008'687'907’852'837'564'279'074' 904'382'605'163'141'518'161'494'336.
+	\end{itemize}
+\vspace{1em}
+
+\uncover<1->{Probability of a hexadecimal hash with certain characteristics:\\
+\vspace{0.5em}
+\tiny 
+$P(h \leq \color{focus}0\color{black}FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF) = \dfrac{1}{16}$\\
+\vspace{1em}
+$P(h \leq \color{focus}00\color{black}FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF) = \left( \dfrac{1}{16} \right)^{2} $
+
+\normalsize}
+
+	
+\end{frame}
+%%%	
+
+%%%
+\begin{frame}{Example in Bitcoin Context}
```

Comment:
I am afraid the example is not correct in this context. In the original lecture, it has been used in the context of Bitcoin addresses. While it is correct that Bitcoin addresses also use (multiple) hash functions,  the derivation process also involves EC math. Please remove the slide.